CONTRIBUTING.md

```diff
@@ -3,19 +3,25 @@
 This project welcomes contributions and suggestions.
 
 ## Developer certificate of origin
+
 Contributions require you to sign a _developer certificate of origin_ (DCO) declaring that you have the right to, and actually do, grant us the rights to use your contribution. For details, visit https://developercertificate.org/.
 
 When you submit a pull request, a DCO-bot will automatically determine whether you need to provide a DCO and decorate the PR appropriately (e.g., label, comment). Simply follow the instructions provided by the bot. You will only need to do this once across all repositories using our DCO.
 
 ## Code of conduct
+
 This project has adopted the [GitHub community guidelines](https://help.github.com/en/github/site-policy/github-community-guidelines).
 
 ## Development process
+
 Development happens against the `master` branch following the [GitHub flow model](https://guides.github.com/introduction/flow/). Contributors create feature branches off of `master`, and their pull requests should target the `master` branch. Maintainers are responsible for prompt review of pull requests.
 
 Pull requests against `master` trigger automated tests that are run through Azure DevOps. Additional test suites are run periodically. When adding new code paths or features, tests are a requirement to complete a pull request. They should be added in the `test` directory.
 
+To build the fairlearn dashboard after making changes to it, [install Yarn](https://yarnpkg.com/lang/en/docs/install), and then run the [widget build script](scripts/build-widget.ps1).
```

Comment:
are people supposed to have windows? I assume ps is for power shell?

Reply:
agreed with @adrinjalali . we should provide linux/unix instructions/scripts too.

Reply:
I'm not going to duplicate scripts. We can write them in python since that's cross-plat, but that'll be a couple days work and I'll leave it up to @vingu to schedule that. powershell exists on linux so this is not a blocker

Reply:
That said, I do agree that it's best not to use powershell or bash or make or whatever your tool of choice is. We should use python.

Reply:
> 
> 
> I'm still not clear on how users are supposed to make from source.
> 
> There should either be enough info in the setup.py that users can do `python setup.py sdist` or `python setup.py bdist`, and do `pip install -e path` from whatever path they're supposed to call it. If the instructions are not as clear, then they could be in some Makefile. Can't windows handle makefiles?
> 
> Or should I try to make it and create new build instructions for people on POSIX platforms?

Nothing special is required for users. In fact the commands you wrote above are all possible, with "path" being "." in the root directory (where setup.py is). You just install fairlearn like any other package. This is just if you change the widget code you need to do extra steps. 

Why do we need makefiles here? That's just one more tool like powershell that people on some platforms have and not on others.

Reply:
Sure. I feel like that's a very microsoft centered position, but given this package is mainly developed by microsoft, it makes sense(?).

I guess I'll post a PR regarding development under linux once I actually start working on the package. I don't think python is a desired tool to do what you'd do in makefiles.

Out of my own curiosity, I checked and saw that scikit-learn, pandas, pytoch, tensorflow, numpy, and scipy (the latter 3 not in their root) have Makefiles.

I thought that's how people make :P

Reply:
What exactly are you referring to? I think not liking tools like powershell, or bash is very much company-agnostic :-) 
I had assumed make to be of a similar flavor but that may just as well be due to my ignorance (and I'm willing to fix my ignorance!).

What I'm trying to express is that I don't want duplication per OS because it leads to outdated scripts. That's usually uncontroversial. I'm not very familiar with makefiles, but if they work on every OS and are generally what people like to use then we should do that, no question!

I'll open an issue and we'll address that shortly. I don't want to club it with this change, though. Does that work for you?

Reply:
... forgot to mention: in various internal projects I've found python to be a fantastic tool for any kind of scripting because it's a lot more robust than bash and people who work on such projects are familiar with python already. 

Regardless, I'm the first person to abandon powershell, so let's get it done :-)

Reply:
voila https://github.com/fairlearn/fairlearn/issues/242